File: app/src/main/java/com/datarockets/mnchkn/activities/dashboard/DashboardPresenterImpl.java
package com.datarockets.mnchkn.activities.dashboard;

import com.datarockets.mnchkn.activities.BasePresenter;
import com.datarockets.mnchkn.models.Player;

import java.util.ArrayList;

public class DashboardPresenterImpl extends BasePresenter implements DashboardPresenter,
        DashboardInteractor.OnLoadPlayerListener,
        DashboardInteractor.OnScreenStatusListener {

    private DashboardView mDashboardView;
    private DashboardInteractor mDashboardInteractor;

    public DashboardPresenterImpl(DashboardView dashboardView,
                                  DashboardInteractor dashboardInteractor) {
        this.mDashboardView = dashboardView;
        this.mDashboardInteractor = dashboardInteractor;
    }

    @Override
    public void updatePlayerListItem(Player player, int position) {
        if (mDashboardView != null) {
            mDashboardInteractor.updatePlayer(player, position, this);
        }
    }

    @Override
    public void onCreate() {
        if (mDashboardView != null) {
            mDashboardInteractor.isScreenShouldBeOn(this);
        }
    }

    @Override
    public void onResume() {
        if (mDashboardView != null) {
            mDashboardInteractor.loadPlayersList(this);
        }
    }

    @Override
    public void onFinished(ArrayList<Player> players) {
        if (mDashboardView != null) {
            mDashboardView.setItems(players);
        }
    }

    @Override
    public void onPlayerUpdated(Player player, int position) {
        if (mDashboardView != null) {
            mDashboardView.updatePlayerData(player, position);
        }
    }

    @Override
    public void onDestroy() {
        if (mDashboardView != null) {
            mDashboardView = null;
        }
    }

    @Override
    public void setGameFinished() {
        if (mDashboardView != null) {
            mDashboardInteractor.setGameFinished();
        }
    }

    @Override
    public void insertStep(Player player) {
        if (mDashboardView != null) {
            mDashboardInteractor.insertStep(player);
        }
    }


    @Override
    public void onKeepScreenOn() {
        if (mDashboardView != null) {
            mDashboardView.keepScreenOn(true);
        }
    }

    @Override
    public void onKeepScreenOff() {
        if (mDashboardView != null) {
            mDashboardView.keepScreenOn(false);
        }
    }

}

File: app/src/main/kotlin/com/datarockets/mnchkn/activities/dashboard/DashboardPresenterImpl.kt
package com.datarockets.mnchkn.activities.dashboard

import com.datarockets.mnchkn.models.Player

import java.util.ArrayList

class DashboardPresenterImpl(private var mDashboardView: DashboardView?,
                             private val mDashboardInteractor: DashboardInteractor) : DashboardPresenter, DashboardInteractor.OnLoadPlayerListener, DashboardInteractor.OnScreenStatusListener {

    override fun updatePlayerListItem(player: Player, position: Int) {
        if (mDashboardView != null) {
            mDashboardInteractor.updatePlayer(player, position, this)
        }
    }

    override fun onCreate() {
        if (mDashboardView != null) {
            mDashboardInteractor.isScreenShouldBeOn(this)
        }
    }

    override fun onResume() {
        if (mDashboardView != null) {
            mDashboardInteractor.loadPlayersList(this)
        }
    }

    override fun onFinished(players: ArrayList<Player>) {
        if (mDashboardView != null) {
            mDashboardView!!.setItems(players)
        }
    }

    override fun onPlayerUpdated(player: Player, position: Int) {
        if (mDashboardView != null) {
            mDashboardView!!.updatePlayerData(player, position)
        }
    }

    override fun onDestroy() {
        if (mDashboardView != null) {
            mDashboardView = null
        }
    }

    override fun setGameFinished() {
        if (mDashboardView != null) {
            mDashboardInteractor.setGameFinished()
        }
    }

    override fun insertStep(player: Player) {
        if (mDashboardView != null) {
            mDashboardInteractor.insertStep(player)
        }
    }


    override fun onKeepScreenOn() {
        if (mDashboardView != null) {
            mDashboardView!!.keepScreenOn(true)
        }
    }

    override fun onKeepScreenOff() {
        if (mDashboardView != null) {
            mDashboardView!!.keepScreenOn(false)
        }
    }

}
